src/components/TaskTracker/Task.tsx

```diff
@@ -1,28 +1,78 @@
-import { useEffect, useState } from "react";
+import { useRef, useEffect, useState } from "react";
 import { FaCheck } from "react-icons/fa";
 import { RiArrowGoBackFill } from "react-icons/ri";
 import { BsThreeDotsVertical } from "react-icons/bs";
 import { Settings } from "./Settings";
 import { useTask, useTimer, useBreakStarted } from "@Store";
 import clsx from "clsx";
+import { ITask } from "@Root/src/interfaces";
 
 // TODO: Remove alerted
 // TODO: Add a blurb/instructions to let users know how to toggle
 
-export const Task = ({ task }) => {
+const onClickOff = callback => {
+  const callbackRef = useRef(); // initialize mutable ref, which stores callback
+  const innerRef = useRef(); // returned to client, who marks "border" element
+
+  // update cb on each render, so second useEffect has access to current value 
+  useEffect(() => { callbackRef.current = callback; });
+
+  useEffect(() => {
+    document.addEventListener("click", handleClick);
+    return () => document.removeEventListener("click", handleClick);
+    function handleClick(e) {
+      if (innerRef.current && callbackRef.current &&
+        !innerRef.current.contains(e.target)
+      ) callbackRef.current(e);
+    }
+  }, []);
+
+  return innerRef; // convenience for client (doesn't need to init ref himself) 
+}
+
+export const Task = ({ task, tasks }) => {
   const [openSettings, setOpenSettings] = useState(false);
-  const { completeTask, toggleInProgressState, alertTask, setPomodoroCounter } =
+  const { removeTask, setCompleted, toggleInProgressState, alertTask, setPomodoroCounter, toggleMenu } =
     useTask();
   const { breakStarted } = useBreakStarted();
   const { timerQueue } = useTimer();
+  const innerRef = onClickOff(() => { toggleMenu(task.id, false); });
 
-  function preventFalseInProgress() {
-    if (task.completed) {
-      return;
-    }
+  const openContextMenu = (event) => {
+    event.preventDefault();
+    toggleMenu(task.id, !task.menuToggled);
+
+    /* This is linear search, however it did not seem to 
+       have any perf impact for 100-200 tasks at a time */
+    tasks.forEach((task_: ITask) => {
+      if (task_.menuToggled)
+        toggleMenu(task_.id, false);
+    });
+  }
+
+  const handleDelete = () => {
+    // FIXME: This should be a modal
```

Comment:
Why should this be a modal > an alert?

Reply:
Ideally it is a modal, primarily because users are able to disable pop up dialogues(alert) in the browser. You would get one of 2 behaviors.

* Not give the user the option to cancel(dialogues disabled)
* Wouldn't work altogether.